presto-main/src/main/java/io/prestosql/operator/DynamicFilterCollectOperator.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.operator.aggregation.TypedSet;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.predicate.TupleDomain;
+import io.prestosql.spi.predicate.ValueSet;
+import io.prestosql.spi.type.Type;
+import io.prestosql.spi.type.TypeUtils;
+import io.prestosql.sql.planner.plan.PlanNodeId;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Verify.verify;
+import static io.prestosql.SystemSessionProperties.getDynamicFilteringMaxPerDriverByteSize;
+import static io.prestosql.SystemSessionProperties.getDynamicFilteringMaxPerDriverValueCount;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * This operator acts as a simple "pass-through" pipe, while saving its input pages.
+ * The collected pages' value are used for creating a run-time filtering constraint (for probe-side table scan in an inner join).
+ * We support only small build-side pages (which should be the case when using "broadcast" join).
+ */
+public class DynamicFilterCollectOperator
```

Comment:
Should we rename this class to `DynamicFilterSourceOperator` ? It will help to avoid confusion with `DynamicFilterCollector` class that we'll need to add for the row filtering changes.

Reply:
Sounds good, fixed and force-pushed https://github.com/prestosql/presto/pull/931/commits/4920e008fbb95337f42b984e6fee9c5d25301543.